sys/include/isrpipe.h

```diff
@@ -41,15 +41,21 @@ typedef struct {
 
 /**
  * @brief   Static initializer for irspipe
+ *
+ * @note The size of the buffer (`sizeof(@p BUF)`) must be a power of two.
+ *
+ * @param[in] BUF           buffer to use as ringbuffer
  */
-#define ISRPIPE_INIT(tsrb_buf) { .mutex = MUTEX_INIT, \
-                                 .tsrb = TSRB_INIT(tsrb_buf) }
+#define ISRPIPE_INIT(BUF) { .mutex = MUTEX_INIT_LOCKED, \
+                                 .tsrb = TSRB_INIT(BUF) }
 
 /**
  * @brief   Initialisation function for isrpipe
  *
+ * @note The size of the buffer (@p bufsize) must be a power of two.
+ *
  * @param[in]   isrpipe     isrpipe object to initialize
- * @param[in]   buf         buffer to use as ringbuffer (must be power of two sized!)
```

Comment:
The requirement is still there for `tsrb_t` that is used here, why drop it?

Reply:
It was not dropped, just moved. A `@note` was added to make the requirement a bit more prominent.